src/vs/workbench/browser/workbench.ts

```diff
@@ -192,8 +194,9 @@ export class Workbench extends Layout {
 		const instantiationService = new InstantiationService(serviceCollection, true);
 
 		// {{SQL CARBON EDIT }}
-		// TODO@Davidshi commandLineService currently has no referents, so force its creation
+		// TODO@Davidshi These services currently have no referents, so force their creation
 		serviceCollection.set(ICommandLineProcessing, instantiationService.createInstance(CommandLineService));
+		serviceCollection.set(IAdsTelemetryService, instantiationService.createInstance(AdsTelemetryService));
```

Comment:
why do these services need to be registered here instead of where we register all our other services?  Is it an initialization ordering thing?

Reply:
Yeah you shouldn't have to do this. CommandLineService is unique in terms of how it works. But The Telemetry service should be just another service.